### ISSUE TITLE
VIVO-1389: Provide full text link to works via oaDOI

### ISSUE THREAD

**Issue by chenejac** [vivo-project/VIVO#2980]
[Mike Conlon](https://github.com/mconlon17) (Migrated from [VIVO-1389](https://vivo-project.atlassian.net/browse/VIVO-1389)) said:

oaDOI provides full text links for 90 million articles.  Provide the means to create links to full text by querying oaDOI for all papers with DOI.

Will require team-based design work to insure high-value function for the VIVO community.

**Comment by hauschke** [comment]
oaDOI became Unpaywall, as far as I know. Which again is now more or less OpenAlex. The idea is still good. It would today translate to

1. article has doi
2. look up doi in openalex
3. get back information if fulltext is available as OA plus link
4. display link dynamically in VIVO

This information can change very quickly. I think it would be better to not integrate this into the graph, but keep it dynamically in the frontend, maybe called via Javascript.

**Comment by chenejac** [comment]
Not sure look up OpenAlex in necessary in this process, might be that Unpaywall api is only needed in this process. For instance for this DOI http://doi.org/10.3390/beverages12060066, here is the Unpaywall api request https://api.unpaywall.org/v2/10.3390/beverages12060066?email=chenejac@gmail.com which is returning 

```
"best_oa_location": {
    "endpoint_id": null,
    "evidence": "deprecated",
    "host_type": "publisher",
    "is_best": true,
    "license": "cc-by",
    "oa_date": "2026-06-01",
    "pmh_id": null,
    "repository_institution": null,
    "updated": "deprecated",
    "url": "https://www.mdpi.com/2306-5710/12/6/66/pdf?version=1780300104",
    "url_for_landing_page": "https://doi.org/10.3390/beverages12060066",
    "url_for_pdf": "https://www.mdpi.com/2306-5710/12/6/66/pdf?version=1780300104",
    "version": "publishedVersion"
  }
```

**Comment by hauschke** [comment]
This is correct. And this API call response is smaller than the equivalent in OpenAlex:

https://api.openalex.org/works?filter=doi:10.3390/beverages12060066 

But OpenAlex and Unpaywall are now basically identical. From https://help.openalex.org/access/unpaywall/:

> Unpaywall is not a separate database. Since the Walden rewrite, Unpaywall records are served from the same OpenAlex data that powers everything else — Unpaywall is a legacy-compatible format over that data, kept stable for the large ecosystem built on it. The OA facts in an Unpaywall record and in an OpenAlex [work’s open_access object](https://help.openalex.org/data/works/open-access/) come from the same pipeline.

Further:

> For new projects, consider the [OpenAlex API](https://help.openalex.org/api/) directly — it exposes the same OA information plus everything else OpenAlex knows (works, authors, sources, topics, and more). The Unpaywall format is best when you’re integrating with tools that already speak it.

It could be more promising regarding API stability to directly jump to OpenAlex and use their open_access object.

**Comment by chenejac** [comment]
Looks to me if we are looking only for the information whether linked full text is open or not, it is better to use Unpaywall. If we are looking for other information as well, then using OpenAlex API has more sense. However, we should also take into account potential pricing for OpenAlex API - https://help.openalex.org/access/example-costs/
